### ISSUE TITLE
Typescript example runs with errors

### ISSUE THREAD

**Issue by Alpvax** [nuxt/nuxt.js#5300]
### Version

[v2.4.5](https://github.com/nuxt.js/releases/tag/v2.4.5)

### Reproduction link

[https://github.com/nuxt/nuxt.js/tree/dev/examples/typescript](https://github.com/nuxt/nuxt.js/tree/dev/examples/typescript)

### Steps to reproduce

Run the app (npm run dev)

### What is expected ?

"Hello from client !" should be posted in the dev console, and the app should run without issues. (I had to change the file to .js because `server` is not defined on the `process` object.

### What is actually happening?

One error and 2 warnings are displayed in the console (although the site displays correctly).

`ReferenceError: [nuxt] Error while mounting app: exports is not defined
    at Module../node_modules/babel-loader/lib/index.js?!./node_modules/ts-loader/index.js?!./node_modules/vue-loader/lib/index.js?!./pages/index.vue?vue&type=script&lang=ts& (index.js:220)`

```[HMR] bundle 'client' has 2 warnings
vendors.app.js:2630 ./components/HelloWorld.vue?vue&type=script&lang=ts& 1:235-238"export 'default' (imported as 'mod') was not found in '-!../node_modules/babel-loader/lib/index.js??ref--3-0!../node_modules/ts-loader/index.js??ref--3-1!../node_modules/vue-loader/lib/index.js??vue-loader-options!./HelloWorld.vue?vue&type=script&lang=ts&'./pages/index.vue?vue&type=script&lang=ts& 1:230-233"export 'default' (imported as 'mod') was not found in '-!../node_modules/babel-loader/lib/index.js??ref--3-0!../node_modules/ts-loader/index.js??ref--3-1!../node_modules/vue-loader/lib/index.js??vue-loader-options!./index.vue?vue&type=script&lang=ts&'```

<!--cmty--><!--cmty_prevent_hook-->
<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8873">#c8873</a>)</em></sub></div>

**Comment by chanlito** [comment]
Getting same error on 2.5

**Comment by takenaka** [comment]
Please set tsconfig.json.
```json:tsconfig.json
{
  "compilerOptions": {
    "target": "esnext",
    "module": "esnext",
    "moduleResolution": "node",
    "lib": [
      "esnext",
      "esnext.asynciterable",
      "dom"
    ],
    "esModuleInterop": true,
    "experimentalDecorators": true,
    "allowJs": true,
    "sourceMap": true,
    "strict": true,
    "noImplicitAny": false,
    "noEmit": true,
    "baseUrl": ".",
    "paths": {
      "~/*": [
        "./*"
      ],
      "@/*": [
        "./*"
      ]
    },
    "types": [
      "@types/node",
      "@nuxt/vue-app"
    ]
  }
}
```

**Comment by chanlito** [comment]
Seems fine to me on 2.5.1

**Comment by kevinmarrec** [comment]
@Alpvax Can you confirm it's not reproducible on 2.5.1 ? Thanks.

**Comment by Alpvax** [comment]
Yes, I will do when I get a chance

**Comment by kevinmarrec** [comment]
@Alpvax great, feel free to close the issue if you're not able to reproduce the issue :)

**Comment by israelroldan** [comment]
Still reproducible in 2.5.1 :(

**Comment by Alpvax** [comment]
@israelroldan I cannot reproduce the issue after updating, are you sure your issue is the same?

**Comment by kevinmarrec** [comment]
I wasn't able to reproduce the issue with `2.5.1` + @chanlito @israelroldan confirmed it's not reproducible, closing the issue then.

@israelroldan Please open a new issue with reproduction if you get another issue.

**Comment by MrJmpl3** [comment]
Version: 2.5.1 , Nuxtjs add `"strict": true` in my custom tsconfig.json in `npm run dev`

```json
{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "es2015",
      "es2016",
      "es2017"
    ],
    "module": "es2015",
    "moduleResolution": "node",
    "experimentalDecorators": true,
    "noImplicitAny": false,
    "noImplicitThis": false,
    "noUnusedLocals": true,
    "strictNullChecks": false,
    "removeComments": true,
    "suppressImplicitAnyIndexErrors": true,
    "allowSyntheticDefaultImports": true,
    "baseUrl": ".",
    "allowJs": true,
    "jsx": "preserve",
    "paths": {
      "~/*": [
        "./*"
      ],
      "@/*": [
        "./*"
      ]
    },
    "esModuleInterop": true,
    "sourceMap": true,
    "noEmit": true,
    "types": [
      "@types/node",
      "@nuxt/vue-app"
    ]
  },
  "exclude": [
    "**/*.spec.ts",
    "node_modules"
  ]
}
```

**Comment by kevinmarrec** [comment]
@MrJmpl3 will be fixed in 2.6, Nuxt won't override your `tsconfig.json` anymore.
